### PR TITLE
feat: message reactions design review feedback changes(SQSERVICES-2057, SQSERVICES-2058)

### PR DESCRIPTION
**Pull request by arjita-mitra**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-2057" title="SQSERVICES-2057" target="_blank">SubtaskSQSERVICES-2057</a>  handle reactions when self deleting messages are turned on
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - handle reactions when self deleting messages are turned on
  - remove background highlight for system messages, correct reaction count format, increase reaction list font size

- The **PR Description**
  - message reactions design review changes
----
##### References
1. https://wearezeta.atlassian.net/browse/SQSERVICES-1734

### CODE REVIEW

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/DownloadButton.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import {t} from 'Util/LocalizerUtil';
+
+export interface DownloadButtonProps {
+  actionId: string;
+  messageFocusedTabIndex: number;
+  onDownloadClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  onKeyPress: (event: React.KeyboardEvent<HTMLButtonElement>) => void;
+  children: React.ReactNode;
+  btnClass?: string;
+  styleObj?: CSSObject;
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/MessageActions/DownloadButton.tsx]: I'll go with **styles**

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageActions.tsx [hunk 2]
```diff
@@ -141,6 +141,8 @@ const MessageActionsMenu: FC<MessageActionsMenuProps> = ({
     [message, toggleActiveMenu],
   );
 
+  const isMsgReactable = message.isReactable();
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageActions.tsx:144]:
```suggestion
  const isReactable = message.isReactable();
```

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageActions.tsx [hunk 3]
```diff
@@ -149,29 +151,34 @@ const MessageActionsMenu: FC<MessageActionsMenuProps> = ({
...
-          handleReactionClick={handleReactionClick}
-        />
-        <ReplyButton
-          actionId={MessageActionsId.REPLY}
-          currentMsgActionName={currentMsgActionName}
-          messageFocusedTabIndex={messageFocusedTabIndex}
-          onReplyClick={handleMessageReply}
-          onKeyPress={handleKeyDown}
-        />
+        {isMsgReactable ? (
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageActions.tsx]: If it's a boolean you don't need to use ternary operator :)

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions/MessageReactions.styles.ts [hunk 4]
```diff
@@ -65,10 +65,19 @@ export const messageReactionButtonTooltipTextLink: CSSObject = {
...
+      color: 'var(--accent-color)',
+      fontSize: 'var(--font-size-base)',
+      letterSpacing: '0.031rem',
+    };
+  }
+  return {
+    color: 'var(--message-reactions-count)',
+    fontSize: 'var(--font-size-base)',
+    letterSpacing: '0.031rem',
+  };
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions/MessageReactions.styles.ts]:
```suggestion
  return {
    color: isActive ? 'var(--accent-color)' : 'var(--message-reactions-count)',
    fontSize: 'var(--font-size-base)',
    letterSpacing: '0.031rem',
  };
```

File: src/script/components/Modals/DetailViewModal/DetailViewModalFooter.tsx [hunk 5]
```diff
@@ -130,23 +138,34 @@ const DetailViewModalFooter: FC<DetailViewModalFooterProps> = ({
...
+      {messageEntity.isDownloadable() && !isMsgReactable && (
+        <DownloadButton
+          actionId={MESSAGE_DOWNLOAD_ID}
+          messageFocusedTabIndex={TabIndex.FOCUSABLE}
+          onDownloadClick={handleAssetDownload}
+          onKeyPress={handleKeyDown}
+          btnClass="detail-view-action-button"
+        >
+          <span className="icon-download" />
+          <span>Download</span>
```
Comment (przemvs) [inline src/script/components/Modals/DetailViewModal/DetailViewModalFooter.tsx]: Translation? 🤔

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/DownloadButton.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+const DownloadButton: FC<DownloadButtonProps> = ({
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/MessageActions/DownloadButton.tsx:32]: I think better if you extends DownloadButtonProps with HTMLButton props, so you'll no pass css there, class name, and you'll can pass another props from button :)

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageActions.tsx (hunk 2 above)
Comment (arjita-mitra) [reply src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageActions.tsx:144]: isMsgReactable is more specific so i would keep it that way

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/DownloadButton.tsx (hunk 6 above)
Comment (arjita-mitra) [reply src/script/components/MessagesList/Message/ContentMessage/MessageActions/DownloadButton.tsx:32]: This is not clear to me. what's the benefit of the change you are proposing?

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageActions.tsx (hunk 2 above)
Comment (przemvs) [reply src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageActions.tsx:144]: This components is fully for message, so I think we don't need to add Msg to every variable :)

**Review by przemvs — APPROVED** [review]
LGTM

**Pull request merged**